sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -988,9 +988,10 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
       serde = rowStorage.serde.orElse(fileStorage.serde).orElse(defaultStorage.serde),
       compressed = false,
       properties = rowStorage.properties ++ fileStorage.properties)
-    // If location is defined, we'll assume this is an external table.
-    // Otherwise, we may accidentally delete existing data.
-    val tableType = if (external || location.isDefined) {
+    // If external is defined, we'll assume this is an external table.
+    // Otherwise, we may consider it as managed table where the data
+    // will be deleted on drop table command.
+    val tableType = if (external) {
```

Comment:
Sorry, @sujith71955 . This will cause a huge behavior change. For me, the current Spark behavior is correct and more robust. If the location is given by users, we had better not remove it.

cc @cloud-fan and @gatorsmile .

Reply:
k, then i will close this PR and JIRA.

Reply:
Thanks for your input.

Reply:
CREATE EXTERNAL TABLE is a legacy syntax that we only keep it for hive compatibility. If you look at the CREATE TABLE syntax for native Spark tables, there is no EXTERNAL keyword.

Reply:
yeah i got the point, also this will make sure we wont delete user data, actually this point has come up because we got some usecases where customer facing issue in migrating the hive jobs to spark because of this behaviourial difference. they dont want to manage table :)

Reply:
Even i am not able to figure out any work around to manage this scenario, except i can advice them to manually go and delete the data .